### ISSUE TITLE
Assign some kind of identifier to rules and allow whitelisting or blacklisting violations

### ISSUE THREAD

**Issue by danielpops** [duo-labs/parliament#13]
We have hundreds of policies, and many of them have benign violations. For example, any policy statement with s3:* on all bucket/object resources outputs:

e.g.

```
                "Statement": [
                    {
                        "Action": "s3:*",
                        "Effect": "Allow",
                        "Resource": [
                            "arn:aws:s3:::the_bucket",
                            "arn:aws:s3:::the_bucket/*",
                        ]
                    }
                ],
```

outputs things about `Job` related actions that I personally don't care about.

```
INVALID - No resources match for s3.CreateJob which requires a resource format of * - {'filepath': None}
INVALID - No resources match for s3:DescribeJob which requires a resource format of arn:*:s3:*:*:job/* for the resource job* - {'filepath': None}
INVALID - No resources match for s3.GetAccountPublicAccessBlock which requires a resource format of * - {'filepath': None}
INVALID - No resources match for s3.ListAllMyBuckets which requires a resource format of * - {'filepath': None}
INVALID - No resources match for s3.ListJobs which requires a resource format of * - {'filepath': None}
INVALID - No resources match for s3.PutAccountPublicAccessBlock which requires a resource format of * - {'filepath': None}
INVALID - No resources match for s3:UpdateJobPriority which requires a resource format of arn:*:s3:*:*:job/* for the resource job* - {'filepath': None}
INVALID - No resources match for s3:UpdateJobStatus which requires a resource format of arn:*:s3:*:*:job/* for the resource job* - {'filepath': None}
```

Currently all I can do is `grep -v ` on substrings, but that's not great. It would be nice if, similar to flake8 or pep8 rules, each violation type had an identifier and they could be whitelisted or  blacklisted.

**Comment by 0xdabbad00** [comment]
This is a good idea.  I would likely do something similar to what I did for CloudMapper where I have a yaml file of the different types of audit checks and some additional info about each one. https://github.com/duo-labs/cloudmapper/blob/master/audit_config.yaml

**Comment by 0xdabbad00** [comment]
This is somewhat resolved in https://github.com/duo-labs/parliament/releases/tag/0.3.1
I don't think this is exactly what you'd like though.  I think I need to break up the finding types further somehow maybe.  I also need to expose the config file so you can configure that.

**Comment by 0xdabbad00** [comment]
I believe this is better resolved now in #22.  

Assume you have a policy like the following:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "TestPol",
            "Effect": "Allow",
            "Action": ["s3:*", "ec2:*"],
            "Resource": ["arn:aws:s3:::test", "arn:aws:s3:::test/*"]
        }
    ]
}
```
Parliament will create a single finding that has a VERY long `detail` section.
```
{
  "issue": "RESOURCE_MISMATCH",
  "title": "No resources match for the given action",
  "severity": "MEDIUM",
  "description": "",
  "detail": [
    {
      "action": "s3:CreateAccessPoint",
      "required_format": "arn:*:s3:*:*:accesspoint/*"
    },
    {
      "action": "s3:CreateJob",
      "required_format": "*"
    },
    {
      "action": "s3:DeleteAccessPoint",
      "required_format": "arn:*:s3:*:*:accesspoint/*"
    },
...
```

That output is also using the new `--json` flag which may make it easier to extract out the info of interest and filter it using jq.

**Comment by 0xdabbad00** [comment]
I've added the ability to filter out findings based on the file they come from, or some limited ability right now of the data within the finding itself.  This functionality is in 0.3.5 and is documented at https://github.com/duo-labs/parliament#custom-config-file

For your issue specifically, you could create a config file with the contents:

```
RESOURCE_MISMATCH:
  ignore_locations:
    actions: "s3:*"
```

And it should ignore all of the findings you are currently seeing.
